src/Microsoft.Extensions.Primitives/StringValues.cs

```diff
@@ -253,6 +254,256 @@ public static StringValues Concat(StringValues values1, StringValues values2)
         }
 
+        public static bool Equals(StringValues objA, StringValues objB)
+        {
+            if (IsNullOrEmpty(objA))
+            {
+                return IsNullOrEmpty(objB);
+            }
+
+            if (objA._values == null)
+            {
+                if (objB._values == null)
+                {
+                    return objA._value == objB._value;
+                }
+
+                switch (objB._values.Length)
+                {
+                    case 0:
+                        return objA._values == null;
+                    case 1:
+                        return objA._value == objB._values[0];
+                    default:
+                        return false;
+                }
+            }
+
+            switch (objA._values.Length)
+            {
+                case 0:
+                    return IsNullOrEmpty(objB);
+
+                case 1:
+                    if (objB._values == null)
+                    {
+                        return objA._values[0] == objB._value;
+                    }
+                    if (objB._values.Length != 1)
+                    {
+                        return false;
+                    }
+                    return objB._values[0] == objA._values[0];
+
+                default:
+                    if (objB._values == null)
+                    {
+                        return false;
+                    }
+                    if (objA._values.Length != objB._values.Length)
+                    {
+                        return false;
+                    }
+                    for (var i = 0; i < objA._values.Length; i++)
+                    {
+                        if (objA._values[i] != objB._values[i])
+                        {
+                            return false;
+                        }
+                    }
+                    return true;
+            }
+        }
+
+        public bool Equals(StringValues other)
+        {
+            return Equals(this, other);
+        }
+
+        public static bool Equals(string objB, StringValues objA)
+        {
+            return Equals(objA, objB);
+        }
+
+        public static bool Equals(StringValues objA, string objB)
+        {
+            if (string.IsNullOrEmpty(objB))
+            {
+                return IsNullOrEmpty(objA);
+            }
+
+            if (IsNullOrEmpty(objA))
+            {
+                return false;
+            }
+
+            if (objA._values == null)
+            {
+                return objA._value == objB;
+            }
+
+            switch (objA._values.Length)
+            {
+                case 1:
+                    return objA._values[0] == objB;
+
+                default:
+                    return false;
+            }
+        }
+
+        public bool Equals(string other)
+        {
+            return Equals(this, other);
+        }
+
+        public static bool Equals(string[] objB, StringValues objA)
```

Comment:
The parameters should be named `left` and `right` per the framework design guidelines.